src/sxmlc.c

```diff
@@ -1877,11 +1877,20 @@ int XMLDoc_parse_file_SAX(const SXML_CHAR* filename, const SAX_Callbacks* sax, v
 
 int XMLDoc_parse_buffer_SAX_len(const SXML_CHAR* buffer, int buffer_len, const SXML_CHAR* name, const SAX_Callbacks* sax, void* user)
 {
-	DataSourceBuffer dsb = { buffer, buffer_len, 0 };
-#ifdef _MSC_VER
-	SAX_Data sd = {0};
-#else
+#ifdef __CODEGEARC__
 	SAX_Data sd;
+	DataSourceBuffer dsb;
+	dsb.buf = buffer;
+	dsb.buf_len = buffer_len;
+	dsb.cur_pos = 0;
+#else
+	DataSourceBuffer dsb = { buffer, buffer_len, 0 };

```

Comment:
I understand this does not compile with RAD Studio? Which C version does the compiler use?

Reply:
In fact this code isn't working with Embarcadero RAD studio XE7. It's the reason for the #ifdef __CODEGEARC__ instead of #ifdef _MSC_VER, as for the other ifdefs. On the other hands I compiled this code without issue with Visual Studio 2019, Code::Blocks and xCode :-)